### ISSUE TITLE
Import Grids: missing confirmation

### ISSUE THREAD

**Issue by klues** [asterics/AsTeRICS-Grid#45]
After importing grids there is no confirmation for success. Especially if the list of existing grids is long and new grids are added at the end of the list, the user cannot quickly see if import was successful.

# Solution
Maybe use the notification area that is already existing for webradio in order to show success or error messages. This maybe also makes sense for login.

**Comment by alexangelov14** [comment]
Hello @ms-mialingvo , I've added a message after the progress bar finishes notifying the user that the import has been successful. Check it out and tell me if you'd like any changes to it.

https://gridbeta-github.asterics-foundation.org/alexangelov14/-45-import-grids-confirmation-message/pr

**Comment by ms-mialingvo** [comment]
Thanks!
So, from the technical point of view it's working for me. Though I don't have a damaged file so I can't fully test.

Visually the message looks like that

<img width="2058" height="1126" alt="Image" src="https://github.com/user-attachments/assets/cdd01a3b-ea6a-45a2-8f3a-da526b953948" />

and three things that catch my eye are:
- the spacing from the top is way smaller than the spacing from the bottom
- there is so much space between the two sentences
- there is in general a lot of white space which is to some degree logical since it's not a lot of text, but still there is no need for the popup to be so wide

klues suggested using the existing notification for webradio, which you didn't do without addressing why not, so why not?
(This is *not* a request to change it to that notification! I prefer your version for reasons, but I want to hear your reason first.)

**Comment by alexangelov14** [comment]
For this type of action (in this case importing) for me it made more sense to have something more central and taking the attention of the user. I imagined it would be something similar in terms of design to the progress bar modal that already exists. That was the other option as well, to just add the notification there directly but wasn't sure if it won's be too crowded that way. Resizing it and making the text closer together are remarks that I will take into consideration and tweak it a bit, if we are going to continue with this design.

**Comment by klues** [comment]
I don't like the big popup, especially because I am forced to wait until it disappears. So you'll frustrate fast users who want to continue with their work and at the same time maybe slow users would like to take more time to read the message.

I think it would be much better to use the existing popup area (tooltip), see e.g. here how to use it: https://github.com/asterics/AsTeRICS-Grid/blob/master/src/vue-components/views/manageGridsView.vue#L484

**Comment by ms-mialingvo** [comment]
> I think it would be much better to use the existing popup area (tooltip), see e.g. here how to use it: https://github.com/asterics/AsTeRICS-Grid/blob/master/src/vue-components/views/manageGridsView.vue#L484

<img width="1137" height="572" alt="Image" src="https://github.com/user-attachments/assets/29ebde46-91f9-4206-8a91-4401ee7a1c99" />


So, I like Alexander's suggestion better because due to the bar also being on the middle of the screen, the visual attention of the user is on the middle of the screen already and having the the confirmation appear there feels more logical to me than having to shift the attention to the bottom right. (And so for the updateallthumbnails it would be imo the better option there too). I don't mind being "forced" to wait given it's short and after all, when that message appears, I've already waited waaay longer for the import, respectively the thumbnails-update ;) , though having an X button to close the message could be useful.
(And then I also agree on Alexander's thought of adding it separately instead of directly to the progress bar.)

**Comment by ms-mialingvo** [comment]
> I imagined it would be something similar in terms of design to the progress bar modal that already exists. 

Aaah, so that's why it is so wide then, it's taking the same width than the bar. Hm. Okay, then I'm less bothered about the width for consistency, but it still bothers me visually that the content is not vertically centered 😅

**Comment by klues** [comment]
However, your suggestions are simply against UI/UX best practices. Forcing the user to wait is not good and other big apps do the same thing with tooltips for confirmation, see e.g. Google Drive:

<img width="1920" height="945" alt="Image" src="https://github.com/user-attachments/assets/6761a0da-b433-4288-9aa5-2abe8105d0f5" />

Never saw any big blocking full screen confirmation on a widely used app with good UI/UX.

**Comment by ms-mialingvo** [comment]
> your suggestions are simply against UI/UX best practices

Okay, I don't know about best practices, just going with what feels better to me. But I'm open to changing it if you and @alexangelov14 agree that it's better practice.

> forcing the user to wait is not good

...which is why I suggested adding an x to Alexander's suggestion.

Your thoughts, @alexangelov14 ?

**Comment by ms-mialingvo** [comment]
> However, your suggestions are simply against UI/UX best practices. Forcing the user to wait is not good 

What are your sources? What I can find out is that the high interaction cost of a full window message is justified when it's for a critical/important action. So, if the thumbnails aren't all updated, that is just a minor annoyance but doesn't impact the performance of the grid and the user can nevertheless go ahead, so a toast message would be fine, plus it's an action that is likely to be done repeatedly by the user. However, if grids of a grid set aren't imported correctly it's an issue that absolutely requires immediate action by the user as it impacts the grid set which is the whole purpose of the app, plus it's an action that will in most cases only happen once or at least seldom. So a full window would be justified here. I therefore stick to my view.

**Comment by klues** [comment]
Show me just one full screen blocking success message in any widely used and well known app anywhere - I don't think you'll find one. For errors it's maybe different and you can argue to have a better visible/blocking error message in this case, but we were talking about the success message / confirmation so far.

EDIT: in the end I don't care and if you and @alexangelov14 like the full screen success in this case better - okay, do it. I don't want to waste time in such minor discussions. And it's an action not often and there fore probably not making so much difference for UI/UX. However, if creating this full-screen confirmation, it would be nice if the component used for it would be somehow general, so that it can show confirmations like this (with button "OK" to confirm and "X" to close) and also can be used to ask questions like "Do you really want to delete user XY?" with buttons "Yes / No" or "Ok / Cancel" - this then could be used to replace the not-so-nice native alert boxes used right now.

**Comment by alexangelov14** [comment]
I agree that it needs to be way more versatile and did changes so thats possible for better reusability in the future. Did some minor visual changes as well so now the spacing is less and it to me at least looks a bit better.

https://gridbeta-github.asterics-foundation.org/alexangelov14/-45-import-grids-confirmation-message/pr

**Comment by klues** [comment]
thanks - I've added a code review to the PR.

**Comment by klues** [comment]
merged the PR, so will be released with next release.

**Comment by klues** [comment]
released with https://github.com/asterics/AsTeRICS-Grid/releases/tag/release-2025-12-18-13.36%2F%2B0100
